src/js/components/Image.js

```diff
@@ -0,0 +1,69 @@
+import classNames from 'classnames/dedupe';
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+import Util from '../utils/Util';
+
+const METHODS_TO_BIND = ['onImageError'];
+
+class Image extends React.Component {
+  constructor() {
+    super(...arguments);
+
+    this.state = {
+      imageErrorCount: 0
+    };
+
+    METHODS_TO_BIND.forEach((method) => {
+      this[method] = this[method].bind(this);
+    });
+  }
+
+  componentWillReceiveProps(nextProps) {
+    // Try again if new urls were provided
+    let image = ReactDOM.findDOMNode(this);
+    image.src = nextProps.src;
+
+    this.setState({imageErrorCount: 0});
+  }
+
+  onImageError(event) {
+    let {props: {fallbackSrc}, state: {imageErrorCount}} = this;
+
+    if (imageErrorCount === 0) {
+      event.target.src = fallbackSrc;
+    }
+
+    // Both src and fallback failed
+    this.setState({imageErrorCount: imageErrorCount + 1});
```

Comment:
1. I need to know whether both src and fallback have failed in order to tell whether to add the class 'hidden' or not.
2. We can add that if we need it. We don't need it now.